pkg/webhook/fleetresourcehandler/fleetresourcehandler_webhook.go

```diff
@@ -55,30 +61,39 @@ type fleetResourceValidator struct {
 // Handle receives the request then allows/denies the request to modify fleet resources.
 func (v *fleetResourceValidator) Handle(ctx context.Context, req admission.Request) admission.Response {
 	namespacedName := types.NamespacedName{Name: req.Name, Namespace: req.Namespace}
+	reqGroupVersion := schema.GroupVersion{Group: req.Resource.Group, Version: req.Resource.Version}
 	var response admission.Response
 	if req.Operation == admissionv1.Create || req.Operation == admissionv1.Update || req.Operation == admissionv1.Delete {
-		switch req.Kind {
-		case crdGVK:
-			klog.V(2).InfoS("handling CRD resource", "GVK", crdGVK, "namespacedName", namespacedName, "operation", req.Operation)
-			response = v.handleCRD(req)
-		case mcGVK:
-			klog.V(2).InfoS("handling Member cluster resource", "GVK", mcGVK, "namespacedName", namespacedName, "operation", req.Operation)
-			response = v.handleMemberCluster(req)
-		case imcGVK:
-			klog.V(2).InfoS("handling Internal member cluster resource", "GVK", imcGVK, "namespacedName", namespacedName, "operation", req.Operation)
-			response = v.handleInternalMemberCluster(ctx, req)
-		case roleGVK:
-			klog.V(2).InfoS("handling Role resource", "GVK", roleGVK, "namespacedName", namespacedName, "operation", req.Operation)
-			response = v.handleRole(req)
-		case roleBindingGVK:
-			klog.V(2).InfoS("handling Role binding resource", "GVK", roleBindingGVK, "namespacedName", namespacedName, "operation", req.Operation)
-			response = v.handleRoleBinding(req)
-		case namespaceGVK:
-			klog.V(2).InfoS("handling namespace resource", "GVK", namespaceGVK, "namespacedName", namespacedName, "operation", req.Operation)
-			response = v.handleNamespace(req)
-		default:
-			klog.V(2).InfoS("resource is not monitored by fleet resource validator webhook", "GVK", req.Kind.String(), "namespacedName", namespacedName, "operation", req.Operation)
-			response = admission.Allowed(fmt.Sprintf("user: %s in groups: %v is allowed to modify resource with GVK: %s", req.UserInfo.Username, req.UserInfo.Groups, req.Kind.String()))
+		msgFormat := fmt.Sprintf("handling %s resource", req.Kind.Kind)
+		// req.Name and req.Namespace for namespace objects are populated and the same for kind:Namespace objects, so setting it to empty.
+		if req.Kind.Kind == "Namespace" {
+			req.Namespace = ""
```

Comment:
Hi Arvind! It's just a readability concern, I feel that better clarity can be achieved by refactoring this part a little bit. The same logic can be achieved by:

Reply:
```
switch {
    case ... // Special GVKs, e.g., CRD, MemberCluster, Namespace => need special/additional handling
    case req.Namespace == "" // We know for sure that at this point it's guaranteed not to be a Namespace resource
                                                // If the req.Namespace field is empty, it's a cluster scoped resource, we do not need
                                                // to process it (if I understand it correctly).
    default (or another case):    // Namespace scoped resources, need to make sure that it's not in the reserved
                                                // namespace, unless the request is from a whitelisted source
}
```